src/edu/csus/ecs/pc2/core/IInternalController.java

```diff
@@ -868,8 +869,9 @@ void submitJudgeRun(Problem problem, Language language, SerializedFile mainFile,
      * @param additionalFiles
      * @param overrideTimeMS
      * @param overrideRunId
+     * @throws Exception if an error occurs while attempting to send the Run to the Server
      */
-    void submitRun(ClientId submitter, Problem problem, Language language, String entry_point, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId);
+    void submitRun(ClientId submitter, Problem problem, Language language, String entry_point, SerializedFile mainSubmissionFile, SerializedFile[] additionalFiles, long overrideTimeMS, long overrideRunId) throws Exception;
```

Comment:
Ditto; seems like this should throw `SubmissionRejecteException`.

Reply:
I just did what eclipse suggested.  I guess if an other exception occurs the submissions was not accepted anyway.